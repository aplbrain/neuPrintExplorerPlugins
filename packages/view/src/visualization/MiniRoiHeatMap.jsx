import React from 'react';
import colormap from 'colormap';
import ColorBox from './ColorBox';

const viridisColorMap = colormap({
  colormap: 'viridis',
  nshades: 101,
  format: 'hex',
  alpha: 1
});

export function RoiHeatMap({ roiInfoObject, roiInfoObjectKey, sumOfValues, listOfRoisToUse }) {
  const type = roiInfoObjectKey;
  const total = Math.max(sumOfValues, 0.01);

  return listOfRoisToUse.map(roi => {
    const percent = roiInfoObject[roi] ? ((roiInfoObject[roi][type] * 1.0) / total) * 100 : 0;
    return (
      <ColorBox
        margin={0}
        width={10}
        height={20}
        backgroundColor={
          roiInfoObject[roi]
            ? viridisColorMap[Math.floor(((roiInfoObject[roi][type] * 1.0) / total) * 100)]
            : viridisColorMap[0]
        }
        key={roi}
        title={`${roi} ${Math.round(percent * 100) / 100}%`}
        text=""
      />
    );
  });
}

export function HeatMapLabels({ roiList }) {
  const styles = {
    margin: '0px',
    width: '10px',
    height: '40px',
    overflow: 'visible',
    display: 'inline-flex',
    flexDirection: 'row',
    whiteSpace: 'nowrap',
    transform: 'rotate(-90deg) translateX(-40px)',
    transformOrigin: 'left top 0',
    fontSize: '10px'
  };
  return roiList.map(roi => (
    <div title={roi} style={styles} key={roi}>
      {roi}
    </div>
  ));
}

export function ColorLegend() {
  const styles = {
    display: 'inline-flex',
    overflow: 'visible',
    flexDirection: 'row'
  };

  return (
    <div style={styles}>
      {'0% '}{' '}
      {viridisColorMap.map((color, index) => {
        const key = `${color}${index}`;
        const divStyles = {
          margin: '0px',
          width: '1px',
          height: '10px',
          backgroundColor: color
        };
        return <div key={key} style={divStyles} />;
      })}
      {' 100%'}
    </div>
  );
}

export default ({ roiList, roiInfoObject, preTotal, postTotal }) => {
  const styles = {
    display: 'flex',
    flexDirection: 'row',
    margin: '5px'
  };

  const text = (
    <div key="labels" style={styles}>
      <HeatMapLabels roiList={roiList} />
    </div>
  );
  const hmPost = (
    <div key="post" style={styles}>
      <RoiHeatMap
        listOfRoisToUse={roiList}
        roiInfoObject={roiInfoObject}
        roiInfoObjectKey="post"
        sumOfValues={postTotal}
      />
      inputs
    </div>
  );
  const hmPre = (
    <div key="pre" style={styles}>
      <RoiHeatMap
        listOfRoisToUse={roiList}
        roiInfoObject={roiInfoObject}
        roiInfoObjectKey="pre"
        sumOfValues={preTotal}
      />
      outputs
    </div>
  );
  return [text, hmPost, hmPre];
};
