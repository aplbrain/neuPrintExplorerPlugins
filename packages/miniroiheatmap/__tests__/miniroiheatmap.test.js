import React from 'react';
import renderer from 'react-test-renderer';
import MiniRoiHeatMap from '../lib/miniroiheatmap';

test('MiniRoiHeatMap can render', () => {
  const roiList = ["ME", "VES", "dACA"];
  const roiInfoObject = {
    ME: {
      pre: 21,
      post: 34
    }
  };
  const raw = <MiniRoiHeatMap roiList={roiList} preTotal={21} postTotal={34} roiInfoObject={roiInfoObject} />;
  const component = renderer.create(raw);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


