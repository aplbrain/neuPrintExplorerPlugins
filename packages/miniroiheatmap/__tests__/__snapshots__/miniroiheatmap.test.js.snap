// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniRoiHeatMap can render 1`] = `
Array [
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "margin": "5px",
      }
    }
  >
    <div
      style={
        Object {
          "display": "inline-flex",
          "flexDirection": "row",
          "fontSize": "10px",
          "height": "40px",
          "margin": "0px",
          "overflow": "visible",
          "transform": "rotate(-90deg) translateX(-40px)",
          "transformOrigin": "left top 0",
          "whiteSpace": "nowrap",
          "width": "10px",
        }
      }
      title="ME"
    >
      ME
    </div>
    <div
      style={
        Object {
          "display": "inline-flex",
          "flexDirection": "row",
          "fontSize": "10px",
          "height": "40px",
          "margin": "0px",
          "overflow": "visible",
          "transform": "rotate(-90deg) translateX(-40px)",
          "transformOrigin": "left top 0",
          "whiteSpace": "nowrap",
          "width": "10px",
        }
      }
      title="VES"
    >
      VES
    </div>
    <div
      style={
        Object {
          "display": "inline-flex",
          "flexDirection": "row",
          "fontSize": "10px",
          "height": "40px",
          "margin": "0px",
          "overflow": "visible",
          "transform": "rotate(-90deg) translateX(-40px)",
          "transformOrigin": "left top 0",
          "whiteSpace": "nowrap",
          "width": "10px",
        }
      }
      title="dACA"
    >
      dACA
    </div>
  </div>,
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "margin": "5px",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fde725",
          "color": "#000",
          "display": "inline-flex",
          "flexDirection": "column",
          "height": "20px",
          "justifyContent": "center",
          "margin": "0px",
          "minHeight": "20px",
          "minWidth": "10px",
          "overflow": "visible",
          "width": "10px",
        }
      }
      title="ME 100%"
    >
      
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#440154",
          "color": "#000",
          "display": "inline-flex",
          "flexDirection": "column",
          "height": "20px",
          "justifyContent": "center",
          "margin": "0px",
          "minHeight": "20px",
          "minWidth": "10px",
          "overflow": "visible",
          "width": "10px",
        }
      }
      title="VES 0%"
    >
      
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#440154",
          "color": "#000",
          "display": "inline-flex",
          "flexDirection": "column",
          "height": "20px",
          "justifyContent": "center",
          "margin": "0px",
          "minHeight": "20px",
          "minWidth": "10px",
          "overflow": "visible",
          "width": "10px",
        }
      }
      title="dACA 0%"
    >
      
    </div>
    inputs*
  </div>,
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "margin": "5px",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fde725",
          "color": "#000",
          "display": "inline-flex",
          "flexDirection": "column",
          "height": "20px",
          "justifyContent": "center",
          "margin": "0px",
          "minHeight": "20px",
          "minWidth": "10px",
          "overflow": "visible",
          "width": "10px",
        }
      }
      title="ME 100%"
    >
      
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#440154",
          "color": "#000",
          "display": "inline-flex",
          "flexDirection": "column",
          "height": "20px",
          "justifyContent": "center",
          "margin": "0px",
          "minHeight": "20px",
          "minWidth": "10px",
          "overflow": "visible",
          "width": "10px",
        }
      }
      title="VES 0%"
    >
      
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#440154",
          "color": "#000",
          "display": "inline-flex",
          "flexDirection": "column",
          "height": "20px",
          "justifyContent": "center",
          "margin": "0px",
          "minHeight": "20px",
          "minWidth": "10px",
          "overflow": "visible",
          "width": "10px",
        }
      }
      title="dACA 0%"
    >
      
    </div>
    outputs*
  </div>,
]
`;
